SixTrack/sixtrack.s

```diff
@@ -2045,6 +2070,61 @@ C     Block with data/fields needed for checkpoint/restart of DYNK
 !hr02&rvv(j)*ejf0v(j)/ejfv(j)*ejf0v(j)/ejfv(j)
       sigmv(j)=sigmv(j)+((((((xv(1,j)*cikve-xv(2,j)*crkve)*strackz(i))* &!hr02
      &rvv(j))*ejf0v(j))/ejfv(j))*ejf0v(j))/ejfv(j)                       !hr02
++cd kickelens
+!            write(*,*) 'MF: in kickelens'
+            select case (elens_type(ix))
+              case (1)
+! ANNULAR: hollow elens with uniform annular profile for collimation
+! Space charge density is:
+! 0     if r < R1
+! Const if R1 < r < R2
+! 0     if r > R2
+! Parameters:
+!   elens_theta_max is the maximum kick in radians
+!   elens_r2 is R2 in mm
+!   elens_r2ovr1 = R2 / R1 (by default, 1.5)
+!   elens_offset_x - x offset
+!   elens_offset_y - y offset
+!   elens_bend_entrance - switch bends on at entrance
+!   elens_bend_exit - switch bends on at exit
+! internal parameters to calculate kick:
+!   xelens = x(proton) + elens_offset_x
+!   yelens = y(proton) + elens_offset_y
+!   rrelens = sqrt(xelens**2+yelens**2)
+!   r1elens = radius R1 [mm]
+!   frrelens = shape function [1/mm]
+
+! kick from ideal annular profile
+! 1) apply offset of e-lens
+                xelens=xv(1,j)+elens_offset_x(ix)
+                yelens=xv(2,j)+elens_offset_y(ix)
+! 2) calculate radius
+                rrelens=sqrt((xelens)**2+(yelens**2)) ! radius p-beam
+                r1elens=elens_r2(ix)/elens_r2ovr1(ix) ! inner radius elens
+! 3) calculate kick
+                if (rrelens.gt.r1elens) then ! rrelens <= r1 -> no kick from elens
+                  if (rrelens.lt.elens_r2(ix)) then ! r1 <= rrelens < r2
+                    frrelens = (elens_r2(ix)/(rrelens**2))*
+     &((((rrelens**2)/(r1elens**2))-1)/(elens_r2ovr1(ix)**2 - 1))
+                  endif
+                  if (rrelens.ge.elens_r2(ix)) then ! r1 < r2 <= rrelens
+                    frrelens = elens_r2(ix)/(rrelens**2)
+                  endif
+                  yv(1,j)=yv(1,j)-elens_theta_max(ix)*frrelens*xelens
+                  yv(2,j)=yv(2,j)-elens_theta_max(ix)*frrelens*yelens
+                endif
+! include bends at entrance and exit of elens
```

Comment:
Since it is not implemented, maybe add an error message (in DATEN) in case the user tries to specify it?